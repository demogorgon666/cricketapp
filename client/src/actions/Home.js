import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
