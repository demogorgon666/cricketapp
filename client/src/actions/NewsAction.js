import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
