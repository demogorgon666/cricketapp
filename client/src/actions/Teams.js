import {
	GET_TEAMS,
	GET_RANKS,
	GET_MATCHBYTEAMID,
	GET_TEAM_BATSMEN,
	GET_TEAM_BOWLERS,
	GET_TEAM_FIXTURES,
	GET_TEAM_STATS
} from "./Types";

import { startLoading, stopLoading } from "./LoadingAction";

import axios from "axios";

export const getTeams = match_type => dispatch => {
	return axios
		.post("http://localhost:5000/cricketalpha/teams", match_type)
		.then(res => {
			dispatch({
				type: GET_TEAMS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getRanks = (gender, ranking) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/rankings/" + gender,
			ranking
		)
		.then(res => {
			dispatch({
				type: GET_RANKS,
				payload: res.data.data
			});
			console.log("from actions", ranking);
		})
		.catch(err => {
			console.log(err);
		});
};

export const getMatch = (team_id, gender, match_type) => dispatch => {
	dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/match/" +
				team_id +
				"/" +
				gender,
			match_type
		)
		.then(res => {
			dispatch(stopLoading());
			dispatch({
				type: GET_MATCHBYTEAMID,
				payload: res.data.data
			});
		})
		.catch(err => {
			dispatch(startLoading());
			console.log(err);
		});
};

export const getTeamBatsmen = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbatsmen/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BATSMEN,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamBowlers = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbowlers/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BOWLERS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

//NIKHIL
export const getFixtures = team_name => dispatch => {
	console.log("get fix from actions", team_name);
	return axios
		.post("http://localhost:5000/cricketalpha/teams/fixtures", team_name)
		.then(res => {
			dispatch({
				type: GET_TEAM_FIXTURES,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamStats = (team_id, gender, type) => dispatch => {
	console.log("getTeamStats action- ", team_id, gender, type);

	// dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/stats/" +
				team_id +
				"/" +
				gender,
			type
		)

		.then(res => {
			// dispatch(stopLoading());
			console.log("getTeamStats Action");
			dispatch({
				type: GET_TEAM_STATS,
				payload: res.data.data
			});
		})
		.catch(err => {
			// dispatch(startLoading());
			console.log(err);
		});
};
