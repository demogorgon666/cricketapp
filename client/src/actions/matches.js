import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

