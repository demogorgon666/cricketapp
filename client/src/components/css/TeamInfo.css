.img-team-player {
  width: 55px;
  height: 55px;
}

.teaminfoloader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #2980b9;
  border-bottom: 16px solid #2980b9;
  width: 120px;
  height: 120px;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
  display: flex;
  /* justify-content: center; */
  margin-left: 80%;
  margin-top: 50%;
}

@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.container-team-details {
  margin-top: 200px;
  margin-left: 2%;
  margin-right: 2%;
}

.grid-class-team-details {
  grid-gap: 10px;
  width: 350px;
  padding: 15px;
  border: 1px solid #707070;
  border-radius: 8px;
  margin: 10px;
}

.p-top-team-details {
  text-align: center;
  width: 98%;
  font-size: 28px;
  color: #2c3e50;
}

.cards-new {
  display: flex;
  margin-left: 1%;
  margin-right: 80%;
  margin-top: 2%;
  margin-bottom: 2%;
}

.p-team-details-name {
  font-size: 20px;
  margin-right: auto;
  margin-bottom: 0px;
}
.p-player-style {
  font-size: 10px;
  margin-top: 2px;
  color: #707070;
}

.h2-team-details-position {
  font-size: 20px;
  margin-left: auto;
}
.p-matches {
  color: #2c3e50;
  font-size: 24px;
}

.cards-new {
  margin: 5px 1000px 15px 15px;
}

.matches-section {
  width: 60%;
  float: left;
}

.top-section {
  width: 40%;
  float: right;
}
/* ----------------------------------------------------------------------------------------- */

.all-recent-matches-box-team {
  display: grid;
  grid-template-columns: 25vw 25vw;
  height: auto;
  border-radius: 8px;
  margin: 10px;
  grid-gap: 15px;
}

.inside-recent-matches-box-team {
  display: flex;
  flex-direction: column;
  /* margin: 5px 0px 10px 0px; */
  width: 100%;
  height: 140px;
  border-radius: 8px;
  border: 1px solid #707070;
  cursor: pointer;
}
.list-info {
  text-align: left;
  display: flex;
  margin: 15px 0px 0px 0px;
}
.list-info:not(:last-child) {
  border-bottom: 1px solid #707070;
}

.inside-recent-matches-box-team .Team-data {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  text-align: left;
  font-size: 10pt;
  margin: 3px;
  margin-left: 15px;
  margin-top: 10px;
}

.inside-recent-matches-box-team .tournamnet-name {
  text-align: left;
  font-size: 14px;
  margin-left: 10pt;
  margin-top: 10pt;
  color: #707070;
}

.inside-recent-matches-box-team .TeamOne-name {
  text-align: left;
  font-size: 12pt;
  font-weight: 500;
  margin-left: 12pt;
}
.inside-recent-matches-box-team .TeamTwo-name {
  text-align: left;
  font-size: 12pt;
  font-weight: 500;
  margin-left: 12pt;
}

.inside-recent-matches-box-team .TeamOne-score {
  text-align: left;
  font-size: 12pt;
  font-weight: 500;
  margin-left: 10pt;
  margin-right: 25pt;
}
.inside-recent-matches-box-team .TeamTwo-score {
  text-align: left;
  font-size: 12pt;
  font-weight: 500;
  margin-left: 10pt;
  margin-right: 25pt;
}

.inside-recent-matches-box-team .winner-name {
  text-align: left;
  font-size: 16px;
  font-weight: 600;
  margin-left: 10px;
  margin-top: 5pt;
}

@media only screen and (max-width: 1024px) {
  .cards-new {
    margin: 0px 425px 10px 15px;
  }

  .all-recent-matches-box-team {
    display: grid;
    grid-template-columns: 25vw 25vw;
    /* width: 90%; */
    height: auto;
    /* border: 1px solid black; */
    border-radius: 8px;
    margin: 20px;
  }
  .inside-recent-matches-box-team {
    display: flex;
    flex-direction: column;
    margin: 4px;
    margin-bottom: 10px;
    width: 95%;
    height: 140px;
    border-radius: 8px;
    border: 1px solid #707070;
  }

  .inside-recent-matches-box-team .Team-data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    text-align: left;
    font-size: 8pt;
    margin: 1px;
    margin-left: 15px;
    margin-top: 10px;
  }

  .inside-recent-matches-box-team .tournamnet-name {
    text-align: left;
    font-size: 14px;
    margin-left: 10pt;
    margin-top: 10pt;
  }

  .inside-recent-matches-box-team .TeamOne-name {
    text-align: left;
    font-size: 12pt;
    font-weight: 500;
    margin-left: 10pt;
  }
  .inside-recent-matches-box-team .TeamTwo-name {
    text-align: left;
    font-size: 12pt;
    font-weight: 500;
    margin-left: 10pt;
  }

  .inside-recent-matches-box-team .TeamOne-score {
    text-align: left;
    font-size: 12pt;
    font-weight: 500;
    margin-left: 8pt;
    margin-right: 15pt;
  }
  .inside-recent-matches-box-team .TeamTwo-score {
    text-align: left;
    font-size: 12pt;
    font-weight: 500;
    margin-left: 8pt;
    margin-right: 15pt;
  }

  .inside-recent-matches-box-team .winner-name {
    text-align: left;
    font-size: 15px;
    font-weight: 600;
    margin-left: 10pt;
    margin-top: 5pt;
  }
}

@media only screen and (max-width: 768px) {
  .cards-new {
    margin: 0px 325px 10px 15px;
  }

  .all-recent-matches-box-team {
    display: grid;
    grid-template-columns: 50vw;
    /* width: 90%; */
    height: auto;
    /* border: 1px solid black; */
    border-radius: 8px;
    margin: 20px;
  }
  .inside-recent-matches-box-team {
    display: flex;
    flex-direction: column;
    margin: 4px;
    margin-bottom: 10px;
    width: 95%;
    height: 140px;
    border-radius: 8px;
    border: 1px solid #707070;
  }
  .inside-recent-matches-box-team:hover {
    width: 98%;
    height: 145px;
  }

  .inside-recent-matches-box-team .Team-data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    text-align: left;
    font-size: 10pt;
    margin: 1px;
    margin-left: 15px;
    margin-top: 10px;
  }

  .inside-recent-matches-box-team .tournamnet-name {
    text-align: left;
    font-size: 14px;
    margin-left: 12pt;
    margin-top: 12pt;
  }

  .inside-recent-matches-box-team .TeamOne-name {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 15pt;
  }
  .inside-recent-matches-box-team .TeamTwo-name {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 15pt;
  }

  .inside-recent-matches-box-team .TeamOne-score {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 30pt;
    margin-right: 50pt;
  }
  .inside-recent-matches-box-team .TeamTwo-score {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 30pt;
    margin-right: 50pt;
  }

  .inside-recent-matches-box-team .winner-name {
    text-align: left;
    font-size: 16px;
    font-weight: 600;
    margin-left: 10px;
    margin-top: 7pt;
  }
}

@media only screen and (max-width: 576px) {
  .container-team-details {
    margin-top: 220px;
  }

  .cards-new {
    margin: 5px 15px 15px 25px;
  }

  .all-recent-matches-box-team {
    display: grid;
    grid-template-columns: 85vw;
    /* width: 90%; */
    height: auto;
    /* border: 1px solid black; */
    border-radius: 8px;
    margin: 20px;
  }
  .inside-recent-matches-box-team {
    display: flex;
    flex-direction: column;
    margin: 4px;
    margin-bottom: 10px;
    width: 95%;
    height: 140px;
    border-radius: 8px;
    border: 1px solid #707070;
  }
  .inside-recent-matches-box-team:hover {
    width: 98%;
    height: 145px;
  }

  .inside-recent-matches-box-team .Team-data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    text-align: left;
    font-size: 10pt;
    margin: 1px;
    margin-left: 15px;
    margin-top: 10px;
  }

  .inside-recent-matches-box-team .tournamnet-name {
    text-align: left;
    font-size: 14px;
    margin-left: 10pt;
    margin-top: 10pt;
  }

  .inside-recent-matches-box-team .TeamOne-name {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 11pt;
  }
  .inside-recent-matches-box-team .TeamTwo-name {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 11pt;
  }

  .inside-recent-matches-box-team .TeamOne-score {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 2pt;
    margin-right: 5px;
  }
  .inside-recent-matches-box-team .TeamTwo-score {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 2pt;
    margin-right: 5px;
  }

  .inside-recent-matches-box-team .winner-name {
    text-align: left;
    font-size: 16px;
    font-weight: 600;
    margin-left: 10px;
    margin-top: 5pt;
  }

  .top-section {
    width: 100%;
    float: left;
  }

  .grid-class-team-details {
    width: auto;
  }
}
