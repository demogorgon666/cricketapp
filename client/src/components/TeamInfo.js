import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/TeamInfo.css";
import "./css/TeamLandingPage.css";
import logo from "./images/dhoni.jpg";
// import Navbar from "../components/common/Navbar";
import { getMatch, getTeamBatsmen, getTeamBowlers } from "../actions/Teams";
// import "./css/SecondaryNavbar.css";

import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class TeamInfo extends Component {
  state = {
    testClick: false,
    odiClick: false,
    t20Click: true
  };
  componentDidMount() {
    console.log("team name is ", this.props.teams);
    let type = {
      match_type: "T20",
      player_country: this.props.teams.team_name
    };
    this.props.getTeamBatsmen(this.props.gender, type);
    this.props.getTeamBowlers(this.props.gender, type);
    console.log(
      "batsman is",
      this.props.getTeamBatsmen(this.props.gender, type)
    );
    this.props.getMatch(this.props.teams.team_id, this.props.gender, type);
  }

  onClickTest = () => {
    this.setState({ testClick: true, odiClick: false, t20Click: false });
    let type = {
      match_type: "Test",
      player_country: this.props.teams.team_name
    };
    this.props.getTeamBatsmen(this.props.gender, type);
    this.props.getTeamBowlers(this.props.gender, type);
    this.props.getMatch(this.props.teams.team_id, this.props.gender, type);
  };

  onClickT20 = () => {
    this.setState({ testClick: false, odiClick: false, t20Click: true });
    let type = {
      match_type: "T20",
      player_country: this.props.teams.team_name
    };
    this.props.getTeamBatsmen(this.props.gender, type);
    this.props.getTeamBowlers(this.props.gender, type);
    this.props.getMatch(this.props.teams.team_id, this.props.gender, type);
  };

  onClickOdi = () => {
    this.setState({ testClick: false, odiClick: true, t20Click: false });
    let type = {
      match_type: "ODI",
      player_country: this.props.teams.team_name
    };
    this.props.getTeamBatsmen(this.props.gender, type);
    this.props.getTeamBowlers(this.props.gender, type);
    this.props.getMatch(this.props.teams.team_id, this.props.gender, type);
  };

  render() {
    console.log("teaminfo props is ", this.props.teams.team_id);
    return (
      <div>
        <div className="container-team-details">
          <p className="p-matches" id="id-p-matches">
            Recent Matches
          </p>
          <div className="centered">
            <div className="cards-new">
              <div
                id="click-t20"
                style={{ borderRadius: "8px 0px 0px 8px" }}
                className={this.state.t20Click ? "cardtest" : "cardt20"}
                onClick={this.onClickT20}
              >
                <p className="p-card" id="id-p-t20">
                  T20
                </p>
              </div>
              <div
                id="click-odi"
                className={this.state.odiClick ? "cardtest" : "cardodi"}
                onClick={this.onClickOdi}
              >
                <p className="p-card" id="id-p-odi">
                  ODI
                </p>
              </div>
              <div
                id="click-test"
                style={{ borderRadius: "0px 8px 8px 0px" }}
                className={this.state.testClick ? "cardtest" : "cardodi"}
                onClick={this.onClickTest}
              >
                <p className="p-card" id="id-p-test">
                  Test
                </p>
              </div>
            </div>
          </div>
          <div className="matches-section">
            <div className="all-recent-matches-box-team">
              {this.props.isLoading ? (
                <div style={{ margin: "0px 0px 0px 80px" }}>
                  <Loader
                    type="TailSpin"
                    color="#2980b9"
                    height="100"
                    width="100"
                  />
                  <span
                    style={{
                      fontSize: "12px"
                    }}
                  >
                    Loading Recent Matches. . .
                  </span>
                </div>
              ) : (
                this.props.matches.map(matches => (
                  <div className="inside-recent-matches-box-team">
                    <span className="tournamnet-name" id="id-match-type">
                      {matches.matchType}
                    </span>
                    <div className="Team-data">
                      <div className="TeamOne-name" id="id-team-one">
                        {matches.teamOne}
                      </div>
                      <div className="TeamOne-score">
                        {matches.teamOneScore}/{matches.teamOneWicket}
                      </div>
                    </div>
                    <div className="Team-data">
                      <div className="TeamTwo-name" id="id-team-two">
                        {matches.teamTwo}
                      </div>
                      <div className="TeamTwo-score">
                        {matches.teamTwoScore}/{matches.teamTwoWicket}
                      </div>
                    </div>
                    <span className="winner-name" id="id-winner">
                      {matches.teamWinner} WON
                    </span>
                  </div>
                ))
                // (this.props.matches.length == 0)
              )}
            </div>
            {/* )} */}
          </div>
          {/* ------------------------------------------------------------------- */}
          <div className="top-section">
            <div className="grid-container-team">
              <div className="grid-container-team-details">
                <div className="grid-class-team-details">
                  <div className="grid-class-topteam-details">
                    <p className="p-top-team-details" id="id-p-top-runs">
                      Top Run Scorers
                    </p>
                    {/* <div> */}
                    {this.props.batsmen.map(batsmen => (
                      // <div>
                      <div className="list-info">
                        {/* <img src={logo} className="img-card" /> */}
                        <img
                          id="pImage"
                          className="playerImage"
                          src={`data:image/jpeg;base64,${batsmen.player_image}`}
                        ></img>
                        <div>
                          <p
                            className="p-team-details-name"
                            id="#id-batsman-name"
                          >
                            {batsmen.player_name}
                          </p>
                          <p className="p-player-style">Right-hand Bat</p>
                        </div>
                        <h2
                          className="h2-team-details-position"
                          id="#id-batsman-score"
                        >
                          {batsmen.player_stats_value}
                        </h2>
                      </div>
                      //   <hr className="hr-team-card" />
                      // </div>
                    ))}
                    {/* </div> */}
                  </div>
                </div>
                <div className="grid-class-team-details">
                  <div className="grid-class-topteam-details">
                    <p className="p-top-team-details" id="id-p-top-wickets">
                      Top Wicket Takers
                    </p>
                    {/* <div> */}
                    {this.props.bowlers.map(bowlers => (
                      // <div>
                      <div className="list-info">
                        {/* <img src={logo} className="img-card" /> */}
                        <img
                          id="pImage"
                          className="playerImage"
                          src={`data:image/jpeg;base64,${bowlers.player_image}`}
                        ></img>
                        <div>
                          <p
                            className="p-team-details-name"
                            id="#id-bowler-name"
                          >
                            {bowlers.player_name}
                          </p>
                          <p className="p-player-style">Right-arm Fast</p>
                        </div>
                        <h2
                          className="h2-team-details-position"
                          id="#id-bowler-score"
                        >
                          {bowlers.player_stats_value}
                        </h2>
                      </div>
                      // <hr className="hr-team-card" />
                      // </div>
                    ))}
                    {/* </div> */}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateTostate = state => ({
  matches: state.TeamsReducer.matches,
  batsmen: state.TeamsReducer.batsmen,
  bowlers: state.TeamsReducer.bowlers,
  isLoading: state.LoadingReducer.isLoading
});

export default connect(
  mapStateTostate,
  { getMatch, getTeamBatsmen, getTeamBowlers }
)(TeamInfo);
