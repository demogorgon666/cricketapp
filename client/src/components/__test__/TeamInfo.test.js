import React from "react";
import { shallow } from "enzyme";
import { TeamInfo } from "../TeamInfo";
import { getMatch, getTeamBatsmen, getTeamBowlers } from "../../actions/Teams";

var matches = [
    {
      matchType: "",
      teamOne: "",
      teamOneScore: "",
      teamOneWicket: "",
      teamTwo: "",
      teamTwoScore: "",
      teamTwoWicket: "",
      teamWinner: ""
    }
  ],
  batsmen = [],
  bowlers = [];

const wrapper = shallow(
  <TeamInfo
    getMatch={getMatch}
    matches={matches}
    getTeamBatsmen={getTeamBatsmen}
    batsmen={batsmen}
    getTeamBowlers={getTeamBowlers}
    bowlers={bowlers}
    match={{ isExact: true, params: { path: "/", url: "/" } }}
  />
);
// const resizeWindow = x => {
//   window.innerWidth = x;
//   // window.innerHeight = y;
//   window.dispatchEvent(new Event("resize"));
// };
describe("Test TeamInfo Component", () => {
  // it("checks for dimensions", () => {
  // const elem = wrapper.find(".inside-recent-matches-box-team");
  // expect(getComputedStyle(elem.getDOMNode()).getPropertyValue('width')).toBe('100%');
  // expect(getComputedStyle(elem.getDOMNode()).getPropertyValue('width')).toBe('%');

  // expect(wrapper.find(".inside-recent-matches-box-team").prop('style')).toHaveProperty(
  //   "width",
  //   "100%"
  // );
  //   resizeWindow(1200);
  //   expect(wrapper.find(".inside-recent-matches-box-team")).toHaveProperty(
  //     "width",
  //     "95%"
  //   );
  // });
  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have links for Test,ODI,T20 for Top Batsmen", () => {
    expect(
      wrapper
        .find("p")
        .at(1)
        .text()
    ).toBe("Top Batsmen");
    expect(
      wrapper
        .find("p")
        .at(2)
        .text()
    ).toBe("Test");

    expect(
      wrapper
        .find("p")
        .at(3)
        .text()
    ).toBe("ODI");

    expect(
      wrapper
        .find("p")
        .at(4)
        .text()
    ).toBe("T20");
  });
  it("should have links for Test,ODI,T20 for Top Bowlers", () => {
    expect(
      wrapper
        .find("p")
        .at(5)
        .text()
    ).toBe("Top Bowlers");
    expect(
      wrapper
        .find("p")
        .at(6)
        .text()
    ).toBe("Test");

    expect(
      wrapper
        .find("p")
        .at(7)
        .text()
    ).toBe("ODI");

    expect(
      wrapper
        .find("p")
        .at(8)
        .text()
    ).toBe("T20");
  });
  it("should check if the tabs are getting clicked ", () => {
    wrapper
      .find("div")
      .at(0)
      .simulate("click");
  });
  it("should have text brand CricketAlpha, matches, teams, players, stats", () => {
    expect(wrapper.find(".secLink").length).toBe(4);
    expect(
      wrapper
        .find(".secLink")
        .at(0)
        .text()
    ).toBe("Info");
    expect(
      wrapper
        .find(".secLink")
        .at(1)
        .text()
    ).toBe("Players");
    expect(
      wrapper
        .find(".secLink")
        .at(2)
        .text()
    ).toBe("Stats");
    expect(
      wrapper
        .find(".secLink")
        .at(3)
        .text()
    ).toBe("Fixtures");
  });
  it("should have className = 'container-team-details' exacty 1", () => {
    expect(wrapper.find(".container-team-details").length).toBe(1);
  });
  it("should have className = 'grid-container-team' exacty 1", () => {
    expect(wrapper.find(".grid-container-team").length).toBe(1);
  });
  it("should have className = 'grid-container-team-details' exacty 1", () => {
    expect(wrapper.find(".grid-container-team-details").length).toBe(1);
  });
  it("should have className = 'grid-class-team-details' exacty 2", () => {
    expect(wrapper.find(".grid-class-team-details").length).toBe(2);
  });
  it("should have className = 'grid-class-topteam-details' exacty 2", () => {
    expect(wrapper.find(".grid-class-topteam-details").length).toBe(2);
  });
  it("should have className = 'centered' exacty 1", () => {
    expect(wrapper.find(".centered").length).toBe(1);
  });
  it("should have className = 'cards-new' exacty 1", () => {
    expect(wrapper.find(".cards-new").length).toBe(1);
  });
  it("should have className = 'cardtest' exacty 1", () => {
    expect(wrapper.find(".cardtest").length).toBe(1);
  });
  it("should have className = 'cardodi' exacty 1", () => {
    expect(wrapper.find(".cardodi").length).toBe(1);
  });
  it("should have className = 'cardt20' exacty 1", () => {
    expect(wrapper.find(".cardt20").length).toBe(1);
  });
  it("should have className = 'p-card' exacty 3", () => {
    expect(wrapper.find(".p-card").length).toBe(3);
  });
  it("should have no br tags", () => {
    expect(wrapper.find("br").length).toBe(0);
  });
  // it("should have className = 'hr-team-card' exacty 4", () => {
  //   expect(wrapper.find(".hr-team-card").length).toBe(4);
  // });
  // it("should have className = 'list' exacty 6", () => {
  //   expect(wrapper.find(".list").length).toBe(4);
  // });
  // it("should have className = 'img-card' exacty 6", () => {
  //   expect(wrapper.find(".img-card").length).toBe(4);
  // });
  // it("should have className = 'p-team-details-name' exacty 6", () => {
  //   expect(wrapper.find(".p-team-details-name").length).toBe(4);
  // });
  // it("should have className = 'p-player-style' exacty 4", () => {
  //   expect(wrapper.find(".p-player-style").length).toBe(4);
  // });
  // it("should have className = 'h2-team-details-position' exacty 4", () => {
  //   expect(wrapper.find(".h2-team-details-position").length).toBe(4);
  // });
  it("should have className = 'p-matches' exacty 1", () => {
    expect(wrapper.find(".p-matches").length).toBe(1);
  });
  // it("should have className = 'all-recent-matches-box' exacty 1", () => {
  //   expect(wrapper.find(".all-recent-matches-box").length).toBe(1);
  // });
  // it("should have className = 'inside-recent-matches-box' exacty 5", () => {
  //   expect(wrapper.find(".inside-recent-matches-box").length).toBe(5);
  // });
  it("should have className = 'tournamnet-name' exacty 5", () => {
    expect(wrapper.find(".tournamnet-name").length).toBe(1);
  });
  it("should have className = 'Team-data' exacty 2", () => {
    expect(wrapper.find(".Team-data").length).toBe(2);
  });
  it("should have className = 'TeamOne-name' exacty 1", () => {
    expect(wrapper.find(".TeamOne-name").length).toBe(1);
  });
  it("should have className = 'TeamOne-score' exacty 1", () => {
    expect(wrapper.find(".TeamOne-score").length).toBe(1);
  });
  it("should have className = 'TeamTwo-name' exacty 1", () => {
    expect(wrapper.find(".TeamTwo-name").length).toBe(1);
  });
  it("should have className = 'TeamTwo-score' exacty 1", () => {
    expect(wrapper.find(".TeamTwo-score").length).toBe(1);
  });
  it("should have className = 'winner-name' exacty 1", () => {
    expect(wrapper.find(".winner-name").length).toBe(1);
  });
});
