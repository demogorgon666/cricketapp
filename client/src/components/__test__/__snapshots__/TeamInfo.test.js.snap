// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TeamInfo Component render the component 1`] = `
<div>
  <div
    className="container-team-details"
  >
    <p
      className="p-matches"
      id="id-p-matches"
    >
      Recent Matches
    </p>
    <div
      className="centered"
    >
      <div
        className="cards-new"
      >
        <div
          className="cardtest"
          id="click-t20"
          onClick={[Function]}
          style={
            Object {
              "borderRadius": "8px 0px 0px 8px",
            }
          }
        >
          <p
            className="p-card"
            id="id-p-t20"
          >
            T20
          </p>
        </div>
        <div
          className="cardodi"
          id="click-odi"
          onClick={[Function]}
        >
          <p
            className="p-card"
            id="id-p-odi"
          >
            ODI
          </p>
        </div>
        <div
          className="cardodi"
          id="click-test"
          onClick={[Function]}
          style={
            Object {
              "borderRadius": "0px 8px 8px 0px",
            }
          }
        >
          <p
            className="p-card"
            id="id-p-test"
          >
            Test
          </p>
        </div>
      </div>
    </div>
    <div
      className="matches-section"
    >
      <div
        className="all-recent-matches-box-team"
      >
        <div
          className="inside-recent-matches-box-team"
        >
          <span
            className="tournamnet-name"
            id="id-match-type"
          >
            ODI
          </span>
          <div
            className="Team-data"
          >
            <div
              className="TeamOne-name"
              id="id-team-one"
            >
              South Africa
            </div>
            <div
              className="TeamOne-score"
            >
              327
              /
              8
            </div>
          </div>
          <div
            className="Team-data"
          >
            <div
              className="TeamTwo-name"
              id="id-team-two"
            >
              Australia
            </div>
            <div
              className="TeamTwo-score"
            >
              296
              /
              10
            </div>
          </div>
          <span
            className="winner-name"
            id="id-winner"
          >
            South Africa
             WON
          </span>
        </div>
      </div>
    </div>
    <div
      className="top-section"
    >
      <div
        className="grid-container-team"
      >
        <div
          className="grid-container-team-details"
        >
          <div
            className="grid-class-team-details"
          >
            <div
              className="grid-class-topteam-details"
            >
              <p
                className="p-top-team-details"
                id="id-p-top-runs"
              >
                Top Run Scorers
              </p>
              <div
                className="list-info"
              >
                <img
                  className="playerImage"
                  id="pImage"
                  src="data:image/jpeg;base64,undefined"
                />
                <div>
                  <p
                    className="p-team-details-name"
                    id="#id-batsman-name"
                  >
                    MS Dhoni
                  </p>
                  <p
                    className="p-player-style"
                  >
                    Right-hand Bat
                  </p>
                </div>
                <h2
                  className="h2-team-details-position"
                  id="#id-batsman-score"
                >
                  10004
                </h2>
              </div>
            </div>
          </div>
          <div
            className="grid-class-team-details"
          >
            <div
              className="grid-class-topteam-details"
            >
              <p
                className="p-top-team-details"
                id="id-p-top-wickets"
              >
                Top Wicket Takers
              </p>
              <div
                className="list-info"
              >
                <img
                  className="playerImage"
                  id="pImage"
                  src="data:image/jpeg;base64,undefined"
                />
                <div>
                  <p
                    className="p-team-details-name"
                    id="#id-bowler-name"
                  >
                    Mohammad Sami
                  </p>
                  <p
                    className="p-player-style"
                  >
                    Right-arm Fast
                  </p>
                </div>
                <h2
                  className="h2-team-details-position"
                  id="#id-bowler-score"
                >
                  15
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
